code/game/objects/items/ego_weapons/he.dm

```diff
@@ -525,6 +525,19 @@
 	attribute_requirements = list(
 							FORTITUDE_ATTRIBUTE = 40
 							)
+	var/in_use
+
+/obj/item/ego_weapon/mini/metal/attack_self(mob/user)
+	if(!CanUseEgo(user) || in_use)
+		return
+	in_use = TRUE
+	if(!do_after(user, 6))
```

Comment:
```suggestion
	if(!do_after(user, 6, src))
```
I believe this would fix the issue of duplicate and remove the requirement for "in_use"